tests/shared/file-extensions.spec.js

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const m = require('mochainon');
+const _ = require('lodash');
+const fileExtensions = require('../../lib/shared/file-extensions');
+
+describe('Shared: fileExtensions', function() {
+
+  describe('.getFileExtensions()', function() {
+
+    _.forEach([
+
+      // No extension
+      {
+        file: 'path/to/filename',
+        extension: []
```

Comment:
Only a minor nit, but these test-case `extension` properties could be renamed to `extensions`, given that they sometimes contain more than one element.
Also, perhaps you should add a `foo.bar.img.gz` testcase?